drivers/serial/uart_stm32.c

```diff
@@ -92,13 +120,38 @@ static inline void uart_stm32_set_baudrate(const struct device *dev,
 		return;
 	}
 
-
 #if HAS_LPUART_1
 	if (IS_LPUART_INSTANCE(UartInstance)) {
+		uint32_t lpuartdiv;
+#ifdef USART_PRESC_PRESCALER
+		uint8_t presc_idx;
+
+		for (presc_idx = 0; presc_idx < ARRAY_SIZE(LPUART_PRESCALER_TAB); presc_idx++) {
+			lpuartdiv = lpuartdiv_calc(clock_rate, presc_idx, baud_rate);
+			if (lpuartdiv >= LPUART_BRR_MIN_VALUE && lpuartdiv <= LPUART_BRR_MASK) {
+				break;
+			}
+		}
+
+		if (presc_idx == ARRAY_SIZE(LPUART_PRESCALER_TAB)) {
+			LOG_ERR("Unable to set %s to %d", dev->name, baud_rate);
+			return;
+		}
+
+		presc_idx = (presc_idx << USART_PRESC_PRESCALER_Pos);
```

Comment:
I would prefer you assign the new shifted value to a new variable : 
* after shift,  it is no more an index, so presc_idx doesn't match any more
* after the shift there is no guarantee it is still an uint8_t. 

So assigning the new value to a new uint32_t variable would be better.

Reply:
fixed